### DISCUSSION TITLE
Prevent user direct download for any public files (e.g. images, fonts)

### DISCUSSION THREAD

**Discussion by ottta** [vercel/next.js#19973] [Help]
Hey, is there any way to **prevent _or_ redirect** if user try to direct download static files in public directory? **e.g.** _images, font files_. I've been use "_redirect_" method on next.config.js, but all static files cannot be download by the app. Reference from this discussion #11899

**Comment by jamesmosier** [comment]
There is no way to prevent this. As Tim mentions [here](https://github.com/vercel/next.js/discussions/11899#discussioncomment-4533), any content you send to a users browser they will be able to download.

**Reply by ottta** [reply]
No way because it is nextjs?

**Reply by jamesmosier** [reply]
No way because that's just the way internet browsers works. There is no mechanism within browsers to prevent certain assets from being accessed by others if you are exposing them via a web server.

**Reply by ottta** [reply]
I have an example for that case, please try these sites [Tokotypefaces](https://www.tokotype.com/retail/maleo) and [Family Type](https://familytype.co/#modernera). When you download font files from network services you'll facing restricted page.

**Reply by jamesmosier** [reply]
This is outside the scope of Next.js. You'll need to setup a custom proxy server with custom rules to do something like this.

https://stackoverflow.com/questions/14173675/how-to-protect-files-from-direct-download
https://stackoverflow.com/questions/10236717/how-to-prevent-a-file-from-direct-url-access

**Reply by ottta** [reply]
Oke thank's @jamesmosier , I'll try and be back to discussion

**Reply by qafariamirhossein** [reply]
We should use a 3rd party.
